java-checks/src/test/files/checks/ImmediateReverseBoxingCheck.java

```diff
@@ -1,16 +1,18 @@
 abstract class A {
-  
+
   public void processInt(String s, int a) {}
   public void processInteger(String s, Integer... a) {}
-  
-  void intBoxingAndUnboxing(int int1, Integer integer1, String string) {
+
+  void intBoxingAndUnboxing(int int1, Integer integer1, String string, Double double1) {
     new Integer(int1).intValue(); // Noncompliant [[sc=5;ec=22]] {{Remove the boxing of "int1".}}
     new Integer(1 + 2).intValue(); // Noncompliant [[sc=5;ec=23]] {{Remove the boxing to "Integer".}}
     Integer.valueOf(int1).intValue(); // Noncompliant
     Integer.valueOf(1 + 2).intValue(); // Noncompliant
     processInt(string, new Integer(int1)); // Noncompliant
     new Integer(int1).toString();
     integer1.intValue();
+    Integer.valueOf(integer1); // Noncompliant [[sc=5;ec=20]] {{Remove the boxing to 'Integer'; The argument is already of type Integer.}}
+    Integer.valueOf(double1); // Compliant
```

Comment:
I would add the case with Long.valueOf (to justify the generalization you made in the code).